File: code.py
# for print the nth fibbonacci number 
# using loops first 
def fibo_term_1(n):
    a = 0
    b = 1
  
    if n <=0:
        print("enter only positive numbers.")
    elif n == 1:
        print(0)
    elif n == 2:
        print(1)
    else:

        for i in range(n-2):
            new = a+b
            a = b
            b = new

        print(new)


# using recursion

def fibo_term_2(n):
    
    if n ==1:
        print(0)
        return
    elif n ==2:
        print(1)
        return
    elif n<=0:
        print("enter only positve numbers.")
    else:
        def fibo(a,b,c):
            if c ==n:
               print(b)
            else:
                return fibo(b,b+a,c+1)          
                        
        return fibo(0,1,2)


# optimised apporch using recursion

def fibo_term_3(n):
    if n<=1:
        return n
    return print(fibo_term_3(n-1) + fibo_term_3(n-2))

fibo_term_1(9)
fibo_term_2(9)
fibo_term_3(9)

File: code.cpp
// c++ code to print the nth fibo term 

// first function for print the nth fibo term using for loop

#include <iostream>
using namespace std;

int fibo_term_1(int n){
    int a = 0;
    int b = 1;
    int new_term = 0;
    if (n ==1) {
        return 0;
    } else if (n ==2) {
        return 1;
    } else if (n>2) {
        for (int i = 0;i<n-2;i++) {
            new_term = a+b;
            a = b;
            b= new_term;
        }
    }
    return new_term;
}
// second funtion using resursion 
// have to use function this time cause c++ does not allow nested function concept 
int fibo(int a,int b,int c ,int n) {
    if (c==n) {
        return b;
    }
    return fibo(b,b+a,c+1,n);
}

int fibo_term_2(int n) {
 
    if (n ==1) {
        return 0;
    } 
    if (n ==2) {
        return 1;
    } 
    if (n<0) {
        cout << "enter only positive number." << endl;
        return -1;

    }
    
    return fibo(0,1,2,n);
}

int fibo_term_3(int n) {
    if(n<=1){
        return 0;
    }
    else {
        return fibo_term_3(n-1) + fibo_term_3(n-2);
    }
}

int main() {
    cout << fibo_term_1(9) << endl;
    cout << fibo_term_2(9) << endl;
    cout << fibo_term_3(4) << endl;
    return 0;
}

